src/herbivoresAndCarnivores.js

```diff
@@ -1,18 +1,52 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100, hidden) {
+    this.health = health;
+    this.name = name;
+    this.hidden = hidden || false;
+    Animal.alive.push(this);
+  }
+
+  static findAnimalByName(name) {
+    return Animal.alive.find(animal => animal.name === name.name);
+  }
+  static removeDeadAnimals() {
+    Animal.alive = Animal.alive.filter(animal => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden) {
+    super(name, health, hidden);
```

Comment:
you see, you do it here)
```suggestion
    super(name, health);
```